middleware/src/middleware.go

```diff
@@ -292,10 +310,10 @@ func (middleware *Middleware) UserChangePassword(args rpcmessages.UserChangePass
 				middleware.dummyIsBaseSetup = true // the change of the admin password completes the dummy setup process (for now)
 			}
 		}
+		return rpcmessages.ErrorResponse{Success: true}
 	}
 
-	errorMessage := "authentication unsuccessful"
-	return rpcmessages.GenericResponse{Success: false, Message: errorMessage}, errors.New(errorMessage)
+	return rpcmessages.ErrorResponse{Success: false, Message: "password change unsuccessful (to short)"}
```

Comment:
You can also get an ErrorResponse when the provided username is not "admin" right? In that case the message would be incorrect.

Reply:
No, the `if len(args.NewPassword) >= 8 {...}` block is closed at this point :)

Reply:
Yep, makes sense. Maybe I looked at a previous commit before or just read it wrong :+1: